vita3k/kernel/src/thread/thread.cpp

```diff
@@ -155,6 +155,9 @@ int start_thread(KernelState &kernel, const SceUID &thid, SceSize arglen, const
 
     const KernelWaitingThreadStates::const_iterator waiting = kernel.waiting_threads.find(thid);
     if (waiting == kernel.waiting_threads.end()) {
+        if (kernel.running_threads.find(thid) != kernel.running_threads.end()) {
+            return SCE_KERNEL_ERROR_THREAD_ALREADY_RUNNING;
```

Comment:
I didn't do any checks but this looks better to me:
```suggestion
            return SCE_KERNEL_ERROR_RUNNING;
```